### PR TITLE
[autoqasm] move pulse module to the top level

### PR DESCRIPTION
**Pull request by yitchen-tim**
*Description of changes:*
 - Rename "pulse_control" -> "pulse"
 - Move pulse module to the top level of autoqasm. This force importing pulses from different path than gates. Separate path also communicates a different level of programming. 
```
from braket.experimental.autoqasm.instructions import rx
from braket.experimental.autoqasm.pulse import capture_v0, play, set_frequency
```

**Reason for not just change the import path without changing the code location:** We would either need to alias or append path of pulse module to the top level. This creates two ways to import pulse, import from aliasing and import from the code location, which is confusing.

*Testing done:*
tox

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#commit-your-change)
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-braket-sdk-python/blob/main/README.md) and [API docs](https://github.com/aws/amazon-braket-sdk-python/blob/main/CONTRIBUTING.md#documentation-guidelines) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/braket/experimental/autoqasm/pulse/__init__.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+"""Pulse programming is a level of programming that controls qubits or frame in a lower level than
+gates, such as playing a waveform or shifting the frequency of a frame. This module includes
+instructions for pulse programming. A program can have only pulse instructions, or mixed usage of
+pulse and gate instructions, if the device supports it. Pulse programming can also used to define
+the calibration of gates in a program.
+
+Example of a program uses only pulse instructions:
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/pulse/__init__.py:20]:
```suggestion
Example of a program that uses only pulse instructions:
```

File: src/braket/experimental/autoqasm/pulse/__init__.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+"""Pulse programming is a level of programming that controls qubits or frame in a lower level than
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/pulse/__init__.py:14]:
```suggestion
"""Pulse programming is a level of programming that controls qubits or frames at a lower level than
```

File: src/braket/experimental/autoqasm/pulse/__init__.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+gates, such as playing a waveform or shifting the frequency of a frame. This module includes
+instructions for pulse programming. A program can have only pulse instructions, or mixed usage of
+pulse and gate instructions, if the device supports it. Pulse programming can also used to define
```
Comment (rmshaffer) [inline src/braket/experimental/autoqasm/pulse/__init__.py:17]:
```suggestion
pulse and gate instructions, if the device supports it. Pulse programming can also be used to define
```

**Review by rmshaffer — APPROVED** [review]
Looks good! Just a few minor doc comments.

File: src/braket/experimental/autoqasm/pulse/__init__.py [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+"""Pulse programming is a level of programming that controls qubits or frames at a lower level than
+gates, such as playing a waveform or shifting the frequency of a frame. This module includes
+instructions for pulse programming. A program can have only pulse instructions, or mixed usage of
+pulse and gate instructions, if the device supports it. Pulse programming can also be used to define
+the calibration of gates in a program.
+
```
Comment (laurencap) [inline src/braket/experimental/autoqasm/pulse/__init__.py:19]: Awesome!

**Review by laurencap — APPROVED** [review]
Awesome, thank you so much! I like this interface!

**Pull request merged**